File: src/cn/vove7/plugin/rest/filetype/RestFileType.java
package cn.vove7.plugin.rest.filetype;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 *
 */
public class RestFileType extends LanguageFileType {

    public final static LanguageFileType INSTANCE = new RestFileType();

    protected RestFileType() {
        super(RestLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Rest file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Rest language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "rest";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return AllIcons.General.Web;
    }
}

File: src/main/kotlin/cn/vove7/plugin/rest/filetype/RestFileType.kt
package cn.vove7.plugin.rest.filetype

import com.intellij.openapi.fileTypes.LanguageFileType
import cn.vove7.plugin.rest.filetype.RestLanguage
import com.intellij.icons.AllIcons
import cn.vove7.plugin.rest.filetype.RestFileType
import javax.swing.Icon

/**
 *
 */
class RestFileType : LanguageFileType(RestLanguage) {
    override fun getName(): String {
        return "Rest file"
    }

    override fun getDescription(): String {
        return "Rest language file"
    }

    override fun getDefaultExtension(): String {
        return "rest"
    }

    override fun getIcon(): Icon {
        return AllIcons.General.Web
    }

    companion object {
        @JvmField
        val INSTANCE: LanguageFileType = RestFileType()
    }
}